controllers/storageclient_controller.go

```diff
@@ -549,6 +561,32 @@ func (r *StorageClientReconciler) reconcileSharedfileStorageClaim() error {
 	return nil
 }
 
+func (r *StorageClientReconciler) reconcileRemoteNoobaa() error {
+	joinSecret := &corev1.Secret{}
+	joinSecret.Name = joinSecretName
+	joinSecret.Namespace = r.OperatorNamespace
+	if err := r.get(joinSecret); err != nil {
+		r.Log.Error(err, "failed to get the joinSecret for remote noobaa")
+		return err
+	}
+	// Create noobaa CR referencing the join secret
+	nb := &nbv1.NooBaa{}
+	nb.Name = "noobaa-remote"
+	nb.Namespace = r.OperatorNamespace
+	nb.Spec.JoinSecret = &corev1.SecretReference{
+		Name:      joinSecretName,
+		Namespace: r.OperatorNamespace,
+	}
+	utils.AddAnnotation(nb, "remote-client-noobaa", "true")
+	if err := r.own(nb); err != nil {
+		return fmt.Errorf("failed to own remote noobaa: %v", err)
+	}
+	if err := r.create(nb); err != nil && !kerrors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create remote noobaa: %v", err)
+	}
+	return nil
```

Comment:
Please use CreateOrUpdate to reconcile the noobaa CR, without it all update operations will go directly into the kube API server and will cause a generation change will in turn requeue another event resulting in an infinite reconcile loop

Reply:
Updated to use CreateOrUpdate